### DISCUSSION TITLE
populateWhere on mongodb

### DISCUSSION THREAD

**Discussion by rubiin** [mikro-orm/mikro-orm#5974] [General]
Is there a way to make populateWhere work on mongodb?

**Comment by B4nan** [comment]
Orovide a repro so it's clear what you want. 

I guess you mean conditions for relations, and nope, since there are no joins in mongo, you cant do that. you need to go the other way around, first load children based on some condition, then their parents, etc.

**Comment by rubiin** [comment]
Currently i am doing something like this

```ts
    const course = await this.courseRepository.findOne(
      { _id, isDeleted: false },
      {
      	populate: ["lesson"]
      },
    );
```

But I wanted to filter the lessons based on the field `isDeleted: false` as well.

Can I do something like this

``` ts
  const course = await this.courseRepository.findOne(
      { _id, isDeleted: false }
    );

  const lessons = await this.lessons.findOne(
      { course: course , isDeleted: false }
    );

course.lessons.add(lessons)

```
Is there a side effect to it?

**Reply by B4nan** [reply]
You can use `populateWhere` for this, I thought you are trying to filter the root entities based on children, this case seems to be simple/possible with mongo too.

Alternatively I would rather load a partial collection then loading separately and adding to the collection.

**Reply by rubiin** [reply]
```ts  
const course = await this.courseRepository.findOne(
      { _id, isDeleted: false },
      {
      	populate: ["lesson","lesson.assignments"]
      },
    ); 
``` 
	
how do we use populateWhere for the `lessons.assigments`
```ts 
 const course = await this.courseRepository.findOne(
      { _id, isDeleted: false },
      {
      	populate: ["lesson","lesson.assignments"],
		populateWhere: {
			lesson : {
			isDeleted: false
			},
			["lesson.assignments"]: {
			isDeleted: false
			},
		}

      },
    ); 
```

This doesnt work @B4nan

**Reply by B4nan** [reply]
> 			["lesson.assignments"]: {
>			isDeleted: false
>			},

This is wrong, you need a nested object, not a dot path (what you do shouldn't even compile, which is always a sign of doing something wrong). Should be something like the following:

```ts
const course = await this.courseRepository.findOne(
  { _id, isDeleted: false },
  {
    populate: ["lesson.assignments"],
    populateWhere: {
      lesson : {
        isDeleted: false,
        assignments: {
          isDeleted: false
        },
      },
    },
  },
); 
```

If that won't help, I need a complete reproduction, just did a quick test and populateWhere seems to be working as expected with mongo too.
